### PR TITLE
[FlagGems Operator Development Competition] add leaky_relu operator

### PR DESCRIPTION
**Pull request by qhy991**
## Summary
- Add `leaky_relu` pointwise operator with Triton kernel.
- Register `leaky_relu` and `leaky_relu_` in FlagGems.
- Add accuracy tests for forward and in-place variants.

## Design / Implementation
- Kernel computes `where(x > 0, x, x * negative_slope)` with FP32 compute.
- Supports configurable `negative_slope` and in-place path.

## Compliance (4.1.2)
- **Style**: PEP 8.
- **API parity**: matches `torch.nn.functional.leaky_relu` and in-place variant.
- **Originality**: implemented for FlagGems competition.
- **License**: agree to Apache 2.0 inclusion with core-contributor attribution.

## Compatibility (4.1.3)
- **Dtypes**: float16, float32, bfloat16.
- **Input dims**: scalar to 5D (POINTWISE_SHAPES).
- **Hardware**: NVIDIA CUDA (user-run).

## Test Coverage Checklist (4.1.4)
- POINTWISE_SHAPES × FLOAT_DTYPES × negative_slope {0.01, 0.1, 2.0}.
- Covers forward and in-place branches.

## Testing
- `python -m pytest tests/test_unary_pointwise_ops.py -k 'leaky_relu' -v`
  - **108 passed**, **0 failed**

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
